shared/elements/gaia_grid/js/items/grid_item.js

```diff
@@ -78,6 +78,14 @@
     persistToDB: true,
 
     /**
+     * Cached values of the last transform of the element, to avoid redundant
+     * style changes.
+     */
+    lastX: -1,
```

Comment:
nit: I wonder if we should use ```null``` instead? I had to stop and think for a second if we could hit the -1 case because you can drag the icon near an edge, and since you drag the center of the icon we might. It's probably fine, but might be more clear code-wise if we default to ```null```.